### DISCUSSION TITLE
Рандомное подключение к голосовым каналам Discord

### DISCUSSION THREAD

**Discussion by FdFilosof** [Anonym-tsk/nfqws-keenetic#91] [Обсуждения]
**Опишите проблему**
Подключение к голосовому каналу Discord происходит рандомно. Заметил зависимость, что после перезагрузки роутера - всегда заходит со 100%-й вероятностью (устанавливает маршрут и можно общаться), спустя какое-то время - абсолютный рандом. Может совершенно не устанавливать соединение, а может делать это без проблем на любом сервере любого региона.
p.s. Интересно, что подобная же история наблюдается с сайтом instagram.com - но на него просто рандомно то заходит - то не заходит. Причём у меня настроен VPN на роутере (чтобы с телефона можно было пользоваться всеми благами домашней сети, так сказать - и там в Инстаграм заходит всегда и безо всяких проблем, что вообще уже странно, как по мне.

**Модель маршрутизатора**
Keenetic Ultra (KN-1811)
Установленная версия 4.1.7

**Провайдер**
ATHome (eth3)

**Выполните команды и приложите их вывод**
`opkg info nfqws-keenetic`
```
Package: nfqws-keenetic
Version: 2.4.1
Depends: iptables, busybox
Conflicts: tpws-keenetic
Status: install user installed
Section: net
Architecture: aarch64-3.10
Size: 121302
Filename: nfqws-keenetic_2.4.1_aarch64-3.10.ipk
Conffiles:
 /opt/etc/nfqws/nfqws.conf b49e99e9938987df7d24e87d8b87f69cf1f39c3ddf0368462c92bf36e8bbe3a5
 /opt/etc/nfqws/user.list 45dc2adaa172b86d73369c6ed12a8a0e648b851b66293b11514c3b1d4bd3fce6
 /opt/etc/nfqws/auto.list e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
 /opt/etc/nfqws/exclude.list 31fbbe06a06e48f9047db40300750b7f11c203a7434fb1a3151f33518ccd9805
Description: NFQWS service
Installed-Time: 1728571688
```

`/opt/etc/init.d/S51nfqws restart`
```
Stopping NFQWS service...
we have 3 user defined desync profile(s) and default low priority profile 0
Loading hostlist /opt/etc/nfqws/auto.list
loading plain text list
Loaded 6 hosts from /opt/etc/nfqws/auto.list
Loading hostlist /opt/etc/nfqws/user.list
loading plain text list
Loaded 137655 hosts from /opt/etc/nfqws/user.list
Loading hostlist /opt/etc/nfqws/auto.list
loading plain text list
Loaded 6 hosts from /opt/etc/nfqws/auto.list
Loading hostlist /opt/etc/nfqws/user.list
loading plain text list
Loaded 137655 hosts from /opt/etc/nfqws/user.list
Loading hostlist /opt/etc/nfqws/exclude.list
loading plain text list
Loaded 26 hosts from /opt/etc/nfqws/exclude.list
Loading hostlist /opt/etc/nfqws/exclude.list
loading plain text list
Loaded 26 hosts from /opt/etc/nfqws/exclude.list
Started NFQWS service
```

`cat /opt/etc/nfqws/nfqws.conf`
```
# Provider network interface, e.g. eth3
# You can specify multiple interfaces separated by space, e.g. ISP_INTERFACE="eth3"
ISP_INTERFACE="eth3"

# All arguments here: https://github.com/bol-van/zapret (search for `nfqws` on the page)
# HTTP(S) strategy
NFQWS_ARGS="--filter-udp=50000-65535 --dpi-desync=fake,tamper --dpi-desync-any-protocol --dpi-desync-cutoff=d4 --dpi-desync-repeats=6 --new --dpi-desync=fake,split2 --dpi-desync-autottl=2 --dpi-desync-split-pos=1 --dpi-desync-fooling=md5sig,badseq --dpi-desync-fake-tls=/opt/etc/nfqws/tls_clienthello.bin --dpi-desync=fake,split --dpi-desync-split-pos=7 --dpi-desync-ttl=12 --dpi-desync-fooling=md5sig,badsum"

# QUIC strategy
NFQWS_ARGS_QUIC="--dpi-desync=fake --dpi-desync-repeats=6 --dpi-desync-cutoff=d4 --dpi-desync-fooling=badsum --dpi-desync-fake-quic=/opt/etc/nfqws/quic_initial.bin"

# auto - automatically detects blocked resources and adds them to the auto.list
NFQWS_EXTRA_ARGS="--hostlist=/opt/etc/nfqws/user.list --hostlist-auto=/opt/etc/nfqws/auto.list --hostlist-auto-debug=/opt/var/log/nfqws.log --hostlist-exclude=/opt/etc/nfqws/exclude.list"

# IPv6 support
IPV6_ENABLED=0

# HTTP support
HTTP_ENABLED=0

# QUIC support
QUIC_ENABLED=1
UDP_PORTS=443,50000:65535

# Syslog logging level (0 - silent, 1 - debug)
LOG_LEVEL=0

NFQUEUE_NUM=200
USER=nobody
CONFIG_VERSION=3
```

`ps | grep nfqws`
```
 4656 nobody   33396 S    /opt/usr/bin/nfqws --daemon --pidfile=/opt/var/run/nfqws.pid --user=nobody --qnum=200 --filt
 4770 root      5976 S    grep nfqws
```

`iptables-save | grep 200`
```
-A POSTROUTING -o eth3 -p udp -m multiport --dports 443,50000:65535 -m connbytes --connbytes 1:8 --connbytes-mode packets --connbytes-dir original -m mark ! --mark 0x40000000/0x40000000 -j NFQUEUE --queue-num 200 --queue-bypass
-A POSTROUTING -o eth3 -p tcp -m tcp --dport 443 -m connbytes --connbytes 1:8 --connbytes-mode packets --connbytes-dir original -m mark ! --mark 0x40000000/0x40000000 -j NFQUEUE --queue-num 200 --queue-bypass
-A _NDM_IPSEC_FORWARD_MANGLE -d 172.20.8.0/23 -p tcp -m tcp --tcp-flags SYN,RST SYN -j TCPMSS --set-mss 1200
-A _NDM_IPSEC_FORWARD_MANGLE -s 172.20.8.0/23 -p tcp -m tcp --tcp-flags SYN,RST SYN -j TCPMSS --set-mss 1200
```

`sysctl net.netfilter.nf_conntrack_checksum`
```
net.netfilter.nf_conntrack_checksum = 0
```

**Comment by EasyKill33** [comment]
С инстой такая же ерунда, с компьютера через браузер сайт не открывается, с телефонов через приложение заходит четко(через браузер instagram.com тоже не грузит), остальное все работает прекрасно, куда копать?

**Comment by shurshick** [comment]
> **Опишите проблему** Подключение к голосовому каналу Discord происходит рандомно. Заметил зависимость, что после перезагрузки роутера - всегда заходит со 100%-й вероятностью (устанавливает маршрут и можно общаться), спустя какое-то время - абсолютный рандом. Может совершенно не устанавливать соединение, а может делать это без проблем на любом сервере любого региона. p.s. Интересно, что подобная же история наблюдается с сайтом instagram.com - но на него просто рандомно то заходит - то не заходит. Причём у меня настроен VPN на роутере (чтобы с телефона можно было пользоваться всеми благами домашней сети, так сказать - и там в Инстаграм заходит всегда и безо всяких проблем, что вообще уже странно, как по мне.
> 
> **Модель маршрутизатора** Keenetic Ultra (KN-1811) Установленная версия 4.1.7
> 
Такой-же роутер. Единственное отличие прошивка стоит из предварительного канала, 4.2.1. Всё прекрасно работает.  В user.list прописаны сервера Discord?

discord-attachments-uploads-prd.storage.googleapis.com
discord-activities.com
discord.design
discord.gifts
discord.store
discord.tools
discordactivities.com
discordmerch.com
discordpartygames.com
discordsays.com
discordsez.com
discord.com
discordapp.net
discordapp.com
discord.gg
discord.app
discord.media
discordcdn.com
discord.dev
discord.new
discord.gift
discordstatus.com
dis.gd
discord.co

**Reply by FdFilosof** [reply]
> > **Опишите проблему** Подключение к голосовому каналу Discord происходит рандомно. Заметил зависимость, что после перезагрузки роутера - всегда заходит со 100%-й вероятностью (устанавливает маршрут и можно общаться), спустя какое-то время - абсолютный рандом. Может совершенно не устанавливать соединение, а может делать это без проблем на любом сервере любого региона. p.s. Интересно, что подобная же история наблюдается с сайтом instagram.com - но на него просто рандомно то заходит - то не заходит. Причём у меня настроен VPN на роутере (чтобы с телефона можно было пользоваться всеми благами домашней сети, так сказать - и там в Инстаграм заходит всегда и безо всяких проблем, что вообще уже странно, как по мне.
> > **Модель маршрутизатора** Keenetic Ultra (KN-1811) Установленная версия 4.1.7
> 
> Такой-же роутер. Единственное отличие прошивка стоит из предварительного канала, 4.2.1. Всё прекрасно работает. В user.list прописаны сервера Discord?
> 
> discord-attachments-uploads-prd.storage.googleapis.com discord-activities.com discord.design discord.gifts discord.store discord.tools discordactivities.com discordmerch.com discordpartygames.com discordsays.com discordsez.com discord.com discordapp.net discordapp.com discord.gg discord.app discord.media discordcdn.com discord.dev discord.new discord.gift discordstatus.com dis.gd discord.co

Да, добавлены, разумеется. Есть ощущение, что тут какая-то зависимость от провайдера - у некоторых как будто дополнительные условия проверки пакетов или что-то в этом духе.

**Comment by balbomush** [comment]
Аналогично, рандомное подключение. Что на 2.4.2, что на 2.4.1. У самого роутер Giga (KN-1011) с версией 4.1.7
Сделал как посоветовал комрад https://github.com/Anonym-tsk/nfqws-keenetic/issues/88#issuecomment-2404893252

Работает сразу только при перезапуске сервиса через 

> /opt/etc/init.d/S51nfqws restart

**Comment by FdFilosof** [comment]
Кажется, в соседней теме была найдена причина всей этой истории: https://github.com/bol-van/zapret/discussions/475#discussioncomment-10955229
Судя по всему, проблема в том, что Кинетик периодически сбрасывает iptables, но как интегрировать предложенный там скрипт в эту сборку - я не понял, знаний не хватает. Если кто-то сумеет - буду благодарен!

**Reply by Anonym-tsk** [reply]
В nfqws-keenetic нет таких проблем

**Comment by Anonym-tsk** [comment]
Обновите nfqws-keenetic

**Comment by FdFilosof** [comment]
Здоровенное спасибо автору - наконец дошли руки всё доковырять окончательно, всё отлично работает, закатал в user.list большой список заблокированных сайтов и всё в порядке. Также работает войс в дискорде и сам дискорд совершенно без каких-либо проблем. Осталась только микро-проблема с инстаграмом - с компа он почему-то очень нестабильно запускается, иногда заходит - иногда не заходит (с телефона в этой же сети всё ок). Полагаю, у моего провайдера какая-то особая ненависть к инстаграму и он заблокирован либо немного другим способом, либо более сложным. При тесте стратегий обхода выдаёт много разных TTL - 6, 7, 12 и проч. Но сделать его работу стабильной ни при одном из них не вышло, а что делать дальше - как будто сложновато для меня, есть ощущение, что я сдамся и просто куплю VPS со временем :)

В любом случае, автор красавчик, закинул немного на кофе <3
